lock.rst

```diff
@@ -0,0 +1,292 @@
+.. index::
+    single: Lock
+
+Dealing with concurrency with Locks
```

Comment:
```suggestion
Dealing with Concurrency with Locks
```

Reply:
why an upper-case?

note: we also use `concurrency` in lower case in `http_client.rst` 

> That's the trick when concurrency is desired

Reply:
The title-case standard we're using says "Every word should be capitialized, except from closed-class words" (https://symfony.com/doc/current/contributing/documentation/standards#english-language-standards). "concurrency" is not a closed class word (but "with" is)

Reply:
thanks for the reference :+1:

Reply:
Thanks for the quick actions :)